lib/dialects/abstract/query-generator.js

```diff
@@ -308,8 +308,15 @@ class QueryGenerator {
 
     const ignoreDuplicates = options.ignoreDuplicates ? this._dialect.supports.inserts.ignoreDuplicates : '';
     const attributes = allAttributes.map(attr => this.quoteIdentifier(attr)).join(',');
-    const returning = this._dialect.supports.returnValues && options.returning ? ' RETURNING *' : '';
     const onConflictDoNothing = options.ignoreDuplicates ? this._dialect.supports.inserts.onConflictDoNothing : '';
+    let returning = '';
+
+    if (Array.isArray(options.returning)) {
+      const fields = options.returning.map(field => this.quoteIdentifier(field)).join(',');
+      returning += ` RETURNING ${fields}`;
```

Comment:
What if the dialect does not support this, i.e., what if `this._dialect.supports.returnValues` is false? Wouldn't this generate a broken query?

You probably want to add a check for `this._dialect.supports.returnValues` here as well, right?